analytics.go

```diff
@@ -20,14 +20,6 @@ const Version = "3.0.0"
 // Endpoint for the Segment API.
 const Endpoint = "https://api.segment.io"
 
-// DefaultContext of message batches.
-var DefaultContext = map[string]interface{}{
-	"library": map[string]interface{}{
-		"name":    "analytics-go",
-		"version": Version,
-	},
-}
```

Comment:
Agreed, plus I'm not comfortable giving users the option to override the library information (we should be source of truth of that). it'd be neat if we could add something even more powerful like the Java library and provide an interceptor mechanism (https://github.com/segmentio/analytics-java/blob/master/analytics/src/main/java/com/segment/analytics/MessageInterceptor.java#L14-L19). This would let users implement coming logic in a single place rather than at every site where they make a tracking call.